CAN'T RUN ON WINDOWS LOCAL MACHINE

Hi, I have followed all the procedures in Readme.md for running the app both with docker and without docker. Everything seems fine, no errors but http://0.0.0.0:8000 in my browser gives "This site can’t be reached".

Also, can it be deployed on VPS? How?

Thanks

![error](https://user-images.githubusercontent.com/47740236/84569180-07639600-ad74-11ea-8dbc-8b215283ff8f.PNG)


what I get by running docker

D:\54vibesApps\splitvibes>docker-compose -f docker-compose.yml -f docker-compose.build.yml -f docker-compose.dev.yml up --build
Creating network "splitvibes_default" with the default driver
Building frontend
Step 1/11 : FROM node:12-buster
 ---> 1f26404403b8
Step 2/11 : RUN mkdir -p /webapp/frontend/assets
 ---> Using cache
 ---> dcd2c92bba4e
Step 3/11 : WORKDIR /webapp/frontend
 ---> Using cache
 ---> c770fde1c276
Step 4/11 : COPY package.json /webapp/frontend/
 ---> Using cache
 ---> 1e21a7699939
Step 5/11 : COPY package-lock.json /webapp/frontend/
 ---> Using cache
 ---> 2a6c9f73402b
Step 6/11 : RUN npm install
 ---> Running in 4bf7f12b9729
npm WARN spleeter-web-frontend@1.0.0 No description
npm WARN spleeter-web-frontend@1.0.0 No repository field.

npm ERR! network timeout at: https://registry.npmjs.org/@babel/helper-plugin-utils/-/helper-plugin-utils-7.8.3.tgz

--- a/.gitignore
+++ b/.gitignore
@@ -36,5 +36,6 @@ env.bak/
 venv.bak/
 
 *.sqlite3
+huey.*
 .coverage
 media/

--- a/README.md
+++ b/README.md
@@ -5,7 +5,7 @@ Spleeter Web is a web application for isolating or removing the vocal, accompani
 
 It is powered by [Spleeter](https://github.com/deezer/spleeter), the awesome source separation library from Deezer that uses deep learning to separate the various components of a song. Spleeter Web uses the pretrained model [`4stems-model`](https://github.com/deezer/spleeter/wiki/3.-Models#pretrained-model), which performs very well on the [*MusDB*](https://sigsep.github.io/datasets/musdb.html) benchmark.
 
-The app uses [Django](https://www.djangoproject.com/) for the backend API, [React](https://reactjs.org/) for the frontend, [PostgreSQL](https://www.postgresql.org/) for the database, and [Huey](https://huey.readthedocs.io/en/latest/)+[Redis](https://redis.io/) for the task queue.
+The app uses [Django](https://www.djangoproject.com/) for the backend API and [React](https://reactjs.org/) for the frontend. [Huey](https://huey.readthedocs.io/en/latest/) is used for the task queue.
 
 ### Features
 - Uses deep neural networks (Spleeter) to separate audio tracks into any combination of their vocal, accompaniment, bass, and drum components
@@ -39,46 +39,57 @@ The app uses [Django](https://www.djangoproject.com/) for the backend API, [Reac
 ## Getting started without Docker
 ### Requirements
 * 4 GB+ of memory (source separation is memory-intensive)
-* Python 3.6 or 3.7
-* Node.JS 12
-* Redis
-* PostgreSQL
-* ffmpeg
+* Python 3.6 or 3.7 ([link](https://www.python.org/downloads/))
+* Node.JS 12 ([link](https://nodejs.org/en/download/))
+* ffmpeg and ffprobe ([link](https://www.ffmpeg.org/download.html))
+    * On macOS, you can install using Homebrew or MacPorts
+    * On Windows, you can follow [this guide](http://blog.gregzaal.com/how-to-install-ffmpeg-on-windows/)
 
 ### Instructions
-1. Configure PostgreSQL and Redis
-
-    In `django_react/settings_dev.py`, update `DATABASES['default']` to point to your PostgreSQL database.
-
-    By default, it assumes you have a database called `spleeter-web` running on `localhost:5432` with the user `spleeter-web` and no password.
-
-    Next, ensure Redis is running on `localhost:6379`.
-
+1. Create Python virtual environment
+    ```sh
+    > python -m venv env
+    # Unix/macOS:
+    > source env/bin/activate
+    # Windows:
+    > .\env\Scripts\activate
+    ```
 2. Install Python dependencies
     ```sh
-    > python3 -m venv env
-    > source env/bin/activate
-    > pip3 install -r requirements.txt
+    (env) > pip install -r requirements.txt
     ```
 3. Install Node dependencies
     ```sh
     > cd frontend
     > npm install
     ```
-4. Start backend and frontend servers (from project directory):
-
-    Set `DJANGO_DEVELOPMENT` to use development setting overrides. By default it will use production values.
-
+4. Start frontend (from `spleeter-web` directory):
     ```sh
+    # Unix/macOS:
     > export DJANGO_DEVELOPMENT=true
-    > npm run dev --prefix frontend & python manage.py runserver 0.0.0.0:8000
+    # Windows:
+    > set DJANGO_DEVELOPMENT=true
+
+    > npm run dev --prefix frontend
     ```
-5. In a separate session, start Huey worker (Redis should be running):
+5. Start backend in separate terminal (from `spleeter-web` directory):
     ```sh
-    > export DJANGO_DEVELOPMENT=true
-    > export HUEY_WORKERS=1
-    > source env/bin/activate
-    > python manage.py run_huey
+    # Unix/macOS:
+    (env) > export DJANGO_DEVELOPMENT=true
+    # Windows:
+    (env) > set DJANGO_DEVELOPMENT=true
+
+    (env) > python manage.py runserver 0.0.0.0:8000
+    ````
+
+6. Start Huey worker in separate terminal (from `spleeter-web` directory):
+    ```sh
+    # Unix/macOS:
+    (env) > export DJANGO_DEVELOPMENT=true
+    # Windows:
+    (env) > set DJANGO_DEVELOPMENT=true
+
+    (env) > python manage.py run_huey
     ```
 6. Launch **Spleeter Web**
 
@@ -90,16 +101,7 @@ By default, **Spleeter Web** uses the local filesystem to store uploaded files a
 
 You can edit `django_react/settings_docker.py` (if using Docker) or `django_react/settings_dev.py` and set `DEFAULT_FILE_STORAGE` to another backend like `'storages.backends.azure_storage.AzureStorage'`.
 
-If using Docker, create an `.env` file in the project root directory with the **django-storages** parameters corresponding to your cloud provider.
-
-Example of `.env` using Azure Storage:
-```
-AZURE_ACCOUNT_KEY=<key>
-AZURE_ACCOUNT_NAME=<account name>
-...
-```
-
-If not using Docker, set the above values as environment variables.
+In the same file, set the storage backend configuration values (`AZURE_ACCOUNT_KEY` and `AZURE_ACCOUNT_NAME` if using Azure).
 
 ## Deploying
 **Spleeter Web** can be deployed on a VPS or a cloud server such as Azure VMs, AWS EC2, DigitalOcean, etc. Deploying to cloud container services like ECS is not yet supported out of the box.
@@ -135,7 +137,7 @@ If not using Docker, set the above values as environment variables.
     > sudo docker-compose -f docker-compose.yml -f docker-compose.build.yml -f docker-compose.prod.yml -f docker-compose.prod.selfhost.yml up --build -d
     ```
 
-    Otherwise:
+    Otherwise if using a storage provider:
     ```sh
     > sudo docker-compose -f docker-compose.yml -f docker-compose.build.yml -f docker-compose.prod.yml up --build -d
     ```

--- a/api/separate.py
+++ b/api/separate.py
@@ -34,20 +34,15 @@ class SpleeterSeparator:
         :param output_path: Path to output file
         :raises e: FFMPEG error
         """
-        try:
-            waveform, _ = self.audio_adapter.load(input_path, sample_rate=self.sample_rate)
-            prediction = self.separator.separate(waveform)
-            out = np.zeros_like(prediction['vocals'])
-            part_count = 0
+        waveform, _ = self.audio_adapter.load(input_path, sample_rate=self.sample_rate)
+        prediction = self.separator.separate(waveform)
+        out = np.zeros_like(prediction['vocals'])
+        part_count = 0
 
-            # Add up parts that were requested
-            for key in prediction:
-                if parts[key]:
-                    out += prediction[key]
-                    part_count += 1
-            out /= part_count
-            self.audio_adapter.save(output_path, out, self.separator._sample_rate, self.audio_format, self.audio_bitrate)
-        except ffmpeg.Error as e:
-            print('stdout:', e.stdout.decode('utf8'))
-            print('stderr:', e.stderr.decode('utf8'))
-            raise e
+        # Add up parts that were requested
+        for key in prediction:
+            if parts[key]:
+                out += prediction[key]
+                part_count += 1
+        out /= part_count
+        self.audio_adapter.save(output_path, out, self.separator._sample_rate, self.audio_format, self.audio_bitrate)

--- a/api/tasks.py
+++ b/api/tasks.py
@@ -40,7 +40,6 @@ def separate_task(processing_track):
         rel_media_path = os.path.join(settings.SEPARATE_DIR, str(processing_track.id), filename)
         rel_path = os.path.join(settings.MEDIA_ROOT, rel_media_path)
         pathlib.Path(directory).mkdir(parents=True, exist_ok=True)
-
         separator = SpleeterSeparator()
 
         parts = {
@@ -75,7 +74,14 @@ def separate_task(processing_track):
             processing_track.save()
         else:
             raise Exception('Error writing to file')
-    except BaseException as error:
+    except FileNotFoundError as error:
+        print(error)
+        print('Please make sure you have FFmpeg and FFprobe installed.')
+        processing_track.status = ProcessedTrack.Status.ERROR
+        processing_track.error = str(error)
+        processing_track.save()
+    except Exception as error:
+        print(error)
         processing_track.status = ProcessedTrack.Status.ERROR
         processing_track.error = str(error)
         processing_track.save()
@@ -129,7 +135,8 @@ def fetch_youtube_audio(source_file, artist, title, link):
             source_file.save()
         else:
             raise Exception('Error writing to file')
-    except BaseException as error:
+    except Exception as error:
+        print(error)
         fetch_task.status = YTAudioDownloadTask.Status.ERROR
         fetch_task.error = str(error)
         fetch_task.save()

--- a/django_react/settings.py
+++ b/django_react/settings.py
@@ -1,5 +1,4 @@
 import os
-import dj_database_url
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
@@ -13,9 +12,12 @@ SECRET_KEY = os.getenv('SECRET_KEY', 'default')
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = False
 
-ALLOWED_HOSTS = [os.getenv('APP_HOST')]
+ALLOWED_HOSTS = [os.getenv('APP_HOST'), '0.0.0.0', '127.0.0.1', 'localhost']
 
 DEFAULT_FILE_STORAGE = 'storages.backends.azure_storage.AzureStorage'
+# OR
+# DEFAULT_FILE_STORAGE = 'django.core.files.storage.FileSystemStorage'
+
 STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'
 
 AZURE_ACCOUNT_KEY = os.getenv('AZURE_ACCOUNT_KEY', '')
@@ -100,7 +102,14 @@ WSGI_APPLICATION = 'django_react.wsgi.application'
 # https://docs.djangoproject.com/en/3.0/ref/settings/#databases
 
 DATABASES = {
-    'default': dj_database_url.config(conn_max_age=600)
+    'default': {
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': 'spleeter-web',
+        'USER': 'postgres',
+        'PASSWORD': 'postgres',
+        'HOST': '127.0.0.1',
+        'PORT': '5432',
+    }
 }
 
 # Internationalization
@@ -120,7 +129,7 @@ HUEY = {
     'results': False,
     'immediate': False,
     'consumer': {
-        'workers': int(os.getenv('HUEY_WORKERS', '2')),
+        'workers': int(os.getenv('HUEY_WORKERS', '1')),
     },
     'url': os.getenv('REDIS_URL', 'redis://localhost:6379')
 }

--- a/django_react/settings_dev.py
+++ b/django_react/settings_dev.py
@@ -1,12 +1,11 @@
 import os
-import dj_database_url
 
 SECRET_KEY = 'default'
 
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = ['localhost', '127.0.0.1', '0.0.0.0']
+ALLOWED_HOSTS = ['0.0.0.0', '127.0.0.1', 'localhost']
 
 # DEFAULT_FILE_STORAGE = 'storages.backends.azure_storage.AzureStorage'
 # OR
@@ -17,7 +16,10 @@ STATICFILES_STORAGE = 'whitenoise.storage.CompressedManifestStaticFilesStorage'
 # https://docs.djangoproject.com/en/3.0/ref/settings/#databases
 
 DATABASES = {
-    'default': dj_database_url.config(default='postgres://spleeter-web@127.0.0.1:5432/spleeter-web', conn_max_age=600)
+    'default': {
+        'ENGINE': 'django.db.backends.sqlite3',
+        'NAME': 'spleeter-web.sqlite3',
+    }
 }
 
 REST_FRAMEWORK = {
@@ -25,4 +27,13 @@ REST_FRAMEWORK = {
         'rest_framework.renderers.JSONRenderer',
         'rest_framework.renderers.BrowsableAPIRenderer',
     ]
+}
+
+HUEY = {
+	'huey_class': 'huey.SqliteHuey',
+    'results': False,
+    'immediate': False,
+    'consumer': {
+        'workers': int(os.getenv('HUEY_WORKERS', '1')),
+    },
 }

--- a/django_react/settings_dev.py
+++ b/django_react/settings_dev.py
@@ -30,7 +30,7 @@ REST_FRAMEWORK = {
 }
 
 HUEY = {
-	'huey_class': 'huey.SqliteHuey',
+    'huey_class': 'huey.SqliteHuey',
     'results': False,
     'immediate': False,
     'consumer': {

--- a/django_react/settings_docker.py
+++ b/django_react/settings_docker.py
@@ -13,7 +13,7 @@ SECRET_KEY = os.getenv('SECRET_KEY', 'default')
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = False
 
-ALLOWED_HOSTS = [os.getenv('APP_HOST', '0.0.0.0')]
+ALLOWED_HOSTS = [os.getenv('APP_HOST', '0.0.0.0', '127.0.0.1', 'localhost')]
 
 DEFAULT_FILE_STORAGE = 'storages.backends.azure_storage.AzureStorage'
 # OR

--- a/django_react/settings_docker.py
+++ b/django_react/settings_docker.py
@@ -13,7 +13,7 @@ SECRET_KEY = os.getenv('SECRET_KEY', 'default')
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = False
 
-ALLOWED_HOSTS = [os.getenv('APP_HOST', '0.0.0.0', '127.0.0.1', 'localhost')]
+ALLOWED_HOSTS = [os.getenv('APP_HOST'), '0.0.0.0', '127.0.0.1', 'localhost']
 
 DEFAULT_FILE_STORAGE = 'storages.backends.azure_storage.AzureStorage'
 # OR

--- a/requirements.txt
+++ b/requirements.txt
@@ -9,6 +9,7 @@ azure-storage-common==2.1.0
 certifi==2020.4.5.1
 cffi==1.14.0
 chardet==3.0.4
+colorama==0.4.3
 coverage==5.1
 cryptography==2.9.2
 decorator==4.4.2
@@ -50,7 +51,7 @@ pylint==2.5.2
 pylint-django==2.0.15
 pylint-plugin-utils==0.6
 python-dateutil==2.8.1
-python-magic==0.4.18
+python-magic-bin==0.4.14
 pytz==2019.3
 redis==3.5.3
 requests==2.23.0
@@ -73,6 +74,6 @@ urllib3==1.25.9
 Werkzeug==1.0.1
 whitenoise==5.1.0
 wrapt==1.11.2
-youtube-dl==2020.6.6
+-e git+https://github.com/JeffreyCA/youtube-dl.git@761f9a2a8f662bec01d7c67ca8a11ed96592a177#egg=youtube_dl
 youtube-title-parse==1.0.0
 zipp==3.1.0

